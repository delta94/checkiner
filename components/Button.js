import React from 'react';

const Button = ({ className, onClick, children }) => {
  return (
    <button className={className} onClick={onClick}>
      {children}
      <style jsx>{`
        button {
          margin: 0;
          padding: 0;
        }
      `}</style>
    </button>
  );
};

export default Button;
