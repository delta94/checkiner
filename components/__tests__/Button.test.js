/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Button from 'components/Button';

describe('With Enzyme', () => {
  const children = <p>Hello World!</p>;
  it('Button renders passed children', () => {
    const button = shallow(<Button>{children}</Button>);

    expect(button.find('p').text()).toEqual('Hello World!');
  });
});

describe('With Snapshot testing', () => {
  it('Button rendered correctly', () => {
    const component = renderer.create(<Button />);
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
