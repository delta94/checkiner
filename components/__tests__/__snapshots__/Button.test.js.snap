// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot testing Button rendered correctly 1`] = `
<button
  className="jsx-3920943040 "
/>
`;
